freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -117,16 +117,22 @@ public Operation synchronizeUsers(String accountId, String actorCrn, Set<String>
         LOGGER.info("Starting operation [{}] with status [{}]", operation.getOperationId(), operation.getStatus());
 
         if (operation.getStatus() == OperationState.RUNNING) {
-            boolean fullSync = userCrnFilter.isEmpty() && machineUserCrnFilter.isEmpty();
-            if (fullSync) {
-                long currentTime = Instant.now().toEpochMilli();
-                stacks.forEach(stack -> {
-                    UserSyncStatus userSyncStatus = userSyncStatusService.getOrCreateForStack(stack);
-                    userSyncStatus.setLastFullSyncStartTime(currentTime);
-                    userSyncStatusService.save(userSyncStatus);
-                });
+            try {
+                boolean fullSync = userCrnFilter.isEmpty() && machineUserCrnFilter.isEmpty();
+                if (fullSync) {
+                    long currentTime = Instant.now().toEpochMilli();
+                    stacks.forEach(stack -> {
+                        UserSyncStatus userSyncStatus = userSyncStatusService.getOrCreateForStack(stack);
+                        userSyncStatus.setLastFullSyncStartTime(currentTime);
+                        userSyncStatusService.save(userSyncStatus);
+                    });
+                }
+                asyncSynchronizeUsers(operation.getOperationId(), accountId, actorCrn, stacks, userCrnFilter, machineUserCrnFilter, fullSync);
+            } catch (RuntimeException e) {
```

Comment:
I would catch Exceeption

Reply:
I catch RuntimeException because there isn't anything in the code that throws a checked exception. I can go either way on this one.

Catch RuntimeException - we narrow the exceptions classes that we catch. if the try code is changed in the future to throw a checked exception then we will know at compile time that we need to handle that.

Catch Exception - the sematics of this block are to clean up if "anything" goes wrong and Exception better represents "anything" than RuntimeException does. As a counterpoint, then what about Throwable? It may make sense here to catch Throwable because we still want to clean up if an Error occurs.

Reply:
I changed these to catch Throwable and attempt to save the operation failure to the database.